Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp

```diff
@@ -345,22 +347,21 @@ ActsExamples::ProcessCode ActsExamples::RootTrajectorySummaryWriter::writeT(
           param[i] = parameter[i];
         }
 
-        res = {param[Acts::eBoundLoc0] - t_d0,
-               param[Acts::eBoundLoc1] - t_z0,
-               Acts::detail::difference_periodic(param[Acts::eBoundPhi], t_phi,
-                                                 static_cast<float>(2 * M_PI)),
-               param[Acts::eBoundTheta] - t_theta,
-               param[Acts::eBoundQOverP] - t_charge / t_p,
-               param[Acts::eBoundTime] - t_time};
-
-        if (boundParam.covariance().has_value()) {
-          const auto& covariance = *boundParam.covariance();
-          for (unsigned int i = 0; i < Acts::eBoundSize; ++i) {
-            error[i] = std::sqrt(covariance(i, i));
-            if (error[i] != 0.0) {
+        if (foundMajorityParticle) {
+          res = {
+              param[Acts::eBoundLoc0] - t_d0,
+              param[Acts::eBoundLoc1] - t_z0,
+              Acts::detail::difference_periodic(param[Acts::eBoundPhi], t_phi,
+                                                static_cast<float>(2 * M_PI)),
+              param[Acts::eBoundTheta] - t_theta,
+              param[Acts::eBoundQOverP] - t_qop,
+              param[Acts::eBoundTime] - t_time};
+
+          if (boundParam.covariance().has_value()) {
+            const auto& covariance = *boundParam.covariance();
+            for (unsigned int i = 0; i < Acts::eBoundSize; ++i) {
+              error[i] = std::sqrt(covariance(i, i));
               pull[i] = res[i] / error[i];
-            } else {
-              pull[i] = std::numeric_limits<double>::quiet_NaN();
```

Comment:
But then the vector sizes are not the same anymore, are they?

Reply:
I think it should be. my goal is to write `NAN` if we don't know the reconstructed params or true params but maybe I messed this up

Reply:
Ok i might have to reread the diff more closely.